starlette/requests.py

```diff
@@ -211,21 +216,24 @@ async def stream(self) -> typing.AsyncGenerator[bytes, None]:
             yield b""
             return
 
-        if self._stream_consumed:
+        if self._stream_state is self._StreamState.disconnected:
+            raise RuntimeError("Client disconnected")
```

Comment:
This wasn't checked or tested but I think it makes sense to check for it

Reply:
Why?

Reply:
It's for this test: https://github.com/encode/starlette/blob/1dad9de11cd6c65a5f937664930051e0d5b20c6a/tests/test_requests.py#L270-L296

Re-reading the linked asgiref issue, I now realize that with Uvicorn at least it's not a problem. Which makes this much more tangential and unnecessary. I'm going to remove this to minimize the changes.